Limber/MIXMockServer.cpp

```diff
@@ -226,15 +225,26 @@ class Server {
             }
         }
 
-        void handleElementReceivedForMIXChannel(IQ::ref iq, std::shared_ptr<ServerFromClientSession> session) {
+        void handleElementReceivedForMIXChannel(Stanza::ref stanza, std::shared_ptr<ServerFromClientSession> session) {
+            auto message = std::dynamic_pointer_cast<Message>(stanza);
+            if (message) {
+                handleMessageReceived(message, session);
+                return;
+            }
+
+            auto iq = std::dynamic_pointer_cast<IQ>(stanza);
+            if (!iq) {
```

Comment:
The `if (message) … if (!iq)` sounds like an if/else if pattern. Could you rewrite it as such, please. That way you can remove the returns and the control flow is more obvious.